core/arch/arm/plat-imx/registers/imx6-crm_regs.h

```diff
@@ -0,0 +1,861 @@
+#define BM_CCM_CCGR5_SDMA	SHIFT_U32(3, BS_CCM_CCGR5_SDMA)
+#define BS_CCM_CCGR5_SPBA	(12)
+#define BM_CCM_CCGR5_SPBA	SHIFT_U32(3, BS_CCM_CCGR5_SPBA)
+#define BS_CCM_CCGR5_SPDIF	(14)
+#define BM_CCM_CCGR5_SPDIF	SHIFT_U32(3, BS_CCM_CCGR5_SPDIF)
+#define BS_CCM_CCGR5_SSI1	(18)
+#define BM_CCM_CCGR5_SSI1	SHIFT_U32(3, BS_CCM_CCGR5_SSI1)
+#define BS_CCM_CCGR5_SSI2	(20)
+#define BM_CCM_CCGR5_SSI2	SHIFT_U32(3, BS_CCM_CCGR5_SSI2)
+#define BS_CCM_CCGR5_SSI3	(22)
+#define BM_CCM_CCGR5_SSI3	SHIFT_U32(3, BS_CCM_CCGR5_SSI3)
+#define BS_CCM_CCGR5_UART	(24)
+#define BM_CCM_CCGR5_UART	SHIFT_U32(3, BS_CCM_CCGR5_UART)
+#define BS_CCM_CCGR5_UART_SERIAL	(26)
+#define BM_CCM_CCGR5_UART_SERIAL	SHIFT_U32(3, BS_CCM_CCGR5_UART_SERIAL)
+
+#define BS_CCM_CCGR6_USBOH3		(0)
+#define BM_CCM_CCGR6_USBOH3		SHIFT_U32(3, BS_CCM_CCGR6_USBOH3)
+#define BS_CCM_CCGR6_USDHC1		(2)
+#define BM_CCM_CCGR6_USDHC1		SHIFT_U32(3, BS_CCM_CCGR6_USDHC1)
+#define BS_CCM_CCGR6_USDHC2		(4)
+#define BM_CCM_CCGR6_USDHC2		SHIFT_U32(3, BS_CCM_CCGR6_USDHC2)
+#define BS_CCM_CCGR6_USDHC3		(6)
+#define BM_CCM_CCGR6_USDHC3		SHIFT_U32(3, BS_CCM_CCGR6_USDHC3)
+#define BS_CCM_CCGR6_USDHC4		(8)
+#define BM_CCM_CCGR6_USDHC4		SHIFT_U32(3, BS_CCM_CCGR6_USDHC4)
+#define BS_CCM_CCGR6_EMI_SLOW		(10)
+#define BM_CCM_CCGR6_EMI_SLOW		SHIFT_U32(3, BS_CCM_CCGR6_EMI_SLOW)
+#define BS_CCM_CCGR6_VDOAXICLK		(12)
+#define BM_CCM_CCGR6_VDOAXICLK		SHIFT_U32(3, BS_CCM_CCGR6_VDOAXICLK)
+
+
+/*
+ * Define Analog Macros and common bits
+ */
+#define BS_CCM_ANALOG_PLL_LOCK			(31)
+#define BM_CCM_ANALOG_PLL_LOCK			\
+			BIT32(BS_CCM_ANALOG_PLL_LOCK)
+#define BS_CCM_ANALOG_PLL_BYPASS		(16)
+#define BM_CCM_ANALOG_PLL_BYPASS		\
+			BIT32(BS_CCM_ANALOG_PLL_BYPASS)
+#define BS_CCM_ANALOG_PLL_BYPASS_CLK_SRC	(14)
+#define BM_CCM_ANALOG_PLL_BYPASS_CLK_SRC	\
+			SHIFT_U32(0x3, BS_CCM_ANALOG_PLL_BYPASS_CLK_SRC)
+#define CCM_ANALOG_PLL_BYPASS_CLK_SRC(clk)	\
+			(SHIFT_U32(clk, BS_CCM_ANALOG_PLL_BYPASS_CLK_SRC) & \
+				BM_CCM_ANALOG_PLL_BYPASS_CLK_SRC)
+#define CCM_ANALOG_PLL_BYPASS_CLK_SRC_CLK24M	(0x0)
+#define CCM_ANALOG_PLL_BYPASS_CLK_SRC_CLK1	(0x1)
+#define CCM_ANALOG_PLL_BYPASS_CLK_SRC_CLK2	(0x2)
+#define CCM_ANALOG_PLL_BYPASS_CLK_SRC_XOR	(0x3)
+
+#define BS_CCM_ANALOG_PLL_ENABLE	(13)
+#define BM_CCM_ANALOG_PLL_ENABLE	BIT32(BS_CCM_ANALOG_PLL_ENABLE)
+#define BS_CCM_ANALOG_PLL_POWERDOWN	(12)
+#define BM_CCM_ANALOG_PLL_POWERDOWN	BIT32(BS_CCM_ANALOG_PLL_POWERDOWN)
+#define BS_CCM_ANALOG_PLL_DIV_SELECT	(0)
+
+/*
+ * Specific Analog bits definition
+ */
+#define BS_CCM_ANALOG_PLL_ARM_PLL_SEL	(19)
+#define BM_CCM_ANALOG_PLL_ARM_PLL_SEL	BIT32(BS_CCM_ANALOG_PLL_ARM_PLL_SEL)
+#define BS_CCM_ANALOG_PLL_ARM_LVDS_24MHZ_SEL	(18)
+#define BM_CCM_ANALOG_PLL_ARM_LVDS_24MHZ_SEL	\
+			BIT32(BS_CCM_ANALOG_PLL_ARM_LVDS_24MHZ_SEL)
+#define BS_CCM_ANALOG_PLL_ARM_LVDS_SEL		(17)
+#define BM_CCM_ANALOG_PLL_ARM_LVDS_SEL		\
+			BIT32(BS_CCM_ANALOG_PLL_ARM_LVDS_SEL)
+#define BM_CCM_ANALOG_PLL_ARM_DIV_SELECT	\
+			SHIFT_U32(0x7F, BS_CCM_ANALOG_PLL_DIV_SELECT)
+
```

Comment:
The file is 850+lines. please no add unused macros. Also this patch is rework register or cleanup, adding this large file is not good.